README.md

```diff
@@ -19,23 +21,149 @@ let data = T::deserialize_from_value(json.into_value()).unwrap();
+
+```json
+{
+  "message": "Json deserialize error: invalid value: integer `1000`, expected u8 at line 6 column 21",
+  "code": "bad_request",
+  "type": "invalid_request",
+  "link": "https://docs.meilisearch.com/errors#bad_request"
+}
+```
+
+##### The message
+
+> Json deserialize error: invalid value: integer `1000`, expected u8 at line 6 column 21
+
+- The message uses the word `u8`, which definitely won't help a user who doesn't know rust or is unfamiliar with types.
+- The location is provided in terms of lines and columns. While this is generally good, when most of your users
+  will read this message in their terminal, it actually doesn't help much.
+
+#### The rest of the payload
+
+Since serde returned this error and we don't know what happened or on which field it happened, the best we
+can do is generate a code `bad_request` that is common for our whole API. We then use this code to generate
+a link to our documentation to help our users. But how do you want to help anyone with an error
+that absolutely every single route of meilisearch can throw?
+
+#### With deserr
+
+```json
+{
+  "message": "Invalid value at `.typoTolerance.minWordSizeForTypos.oneTypo`: value: `1000` is too large to be deserialized, maximum value authorized is `255`",
+  "code": "invalid_settings_typo_tolerance",
+  "type": "invalid_request",
+  "link": "https://docs.meilisearch.com/errors#invalid-settings-typo-tolerance"
+}
+```
+
+##### The message
+
+> Invalid value at `.typoTolerance.minWordSizeForTypos.oneTypo`: value: `1000` is too large to be deserialized, maximum value authorized is `255`
+
+- We get a more human-readable location; `.typoTolerance.minWordSizeForTypos.oneTypo`. It gives us the faulty field.
+- We also get a non-confusing and helpful message this time; it explicitly tells us that the maximum value authorized is `255`.
+
+##### The rest of the payload
+
+Since deserr called one of our functions in the process, we were able to use a custom error code + link to redirect
+our user to the documentation specific to this feature and this field.
+
+#### More possibilities with deserr that were impossible with serde
+
+##### Adding constraints on multiples fields
+
+In meilisearch, there is another constraint on this `minWordSizeForTypos`, the `twoTypo` field **must be** greater than
+the `oneType` field.
+
+Serde doesn't provide any feature to do that. You could write your own implementation of `Deserialize` for the
+entire sub-object `minWordSizeForTypos`, but that's generally hard, and we tend to avoid doing it.
+Thus, that's the kind of thing you're going to check by hand in your code later on. Which may bring
+inconsistencies between most of the deserialization error messages and your error message.
+
+With deserr, we provide attributes that allow you to validate your structure once it's deserialized.
+
+##### When a field is missing
+
+It's possible to provide your own function when a field is missing.
+
+```rust,ignore
```

Comment:
This could, however, confuse users when they implement it like this and then get a compiler error because they didn't use the `error` attribute in the derive macro.
So I kind of changed my mind now, the current version is also good.

Reply:
I think that's the kind of thing we should show in the `examples` directory but not directly in the README.
And for the README I would prefer if we keep it generic and easy to copy-paste :thinking: